src/GregClient/AuthProviders/IAuthProvider.cs

```diff
@@ -11,6 +11,7 @@ public interface IAuthProvider
 
         LoginState LoginState { get; }
         string Username { get; }
+        //TODO ideally would not force the use of restsharp just to set auth.
```

Comment:
Why not remove `SignRequest`?

Reply:
I decided after much going back and forth that I would just leave the authentication bits alone. We still need to provide *some interface* for authorizing a request which will still require actually passing in a request object of some kind.

SignRequest is the wrong name, but this way clients don't have to touch their interface implementations at all.

I'm more concerned about getting RestSharp updated and working with a new serialization library (system.text.json)

IF we find that we cannot get around conflicts with RestSharp then we'll need to come back here anyway.

So I guess the short answer is, let's leave this improvement for later if RestSharp integrates fine, or sooner if we run into more conflict trouble.